src/backend/src/services/tasks.services.ts

```diff
@@ -165,10 +169,14 @@ export default class TasksService {
    */
   static async editTaskStatus(user: User, taskId: string, status: Task_Status) {
     // Get the original task and check if it exists
-    const originalTask = await prisma.task.findUnique({ where: { taskId }, include: { wbsElement: true } });
+    const originalTask = await prisma.task.findUnique({ where: { taskId }, include: { assignees: true, wbsElement: true } });
     if (!originalTask) throw new NotFoundException('Task', taskId);
     if (originalTask.dateDeleted) throw new DeletedException('Task', taskId);
 
+    if (status === 'IN_PROGRESS' && (!originalTask.deadline || originalTask.assignees.length === 0)) {
```

Comment:
I think you did the opposite of what I said before, we dont need to check the original task, just the new one

Reply:
But what if they are moving the original task into in progress, so the new status is in progress, but the original task doesn't have assignees or a deadline. They shouldn't be able to do that right?

Reply:
you right